Training6/training-6-sagas-login/src/index.js

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+import App from "./App";
+import { createStore, applyMiddleware, compose } from "redux";
+import createSagaMiddleware from "redux-saga";
+import { Provider } from "react-redux";
+import rootReducer from "./reducers/index.js";
+import rootSaga from "./sagas/index";
+import { persistStore } from "redux-persist";
+import { PersistGate } from "redux-persist/integration/react";
+
+const sagaMiddleware = createSagaMiddleware();
+const reduxDevTools =
+  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
+
+let store = createStore(
+  rootReducer,
+  compose(applyMiddleware(sagaMiddleware), reduxDevTools)
+);
+
+const persistor = persistStore(store);
+sagaMiddleware.run(rootSaga);
+
+ReactDOM.render(
+  <Provider store={store}>
+    <PersistGate persistor={persistor}>
```

Comment:
missing define loading component in PersistGate, when the app is not persisted

Reply:
i completely fix it.